driver-core/src/main/com/mongodb/connection/CommandMessage.java

```diff
@@ -126,7 +127,7 @@ protected EncodingMetadata encodeMessageBodyWithMetadata(final BsonOutput bsonOu
                 bsonOutput.writeInt32(0);         // size
                 bsonOutput.writeCString(payload.getPayloadName());
                 writePayload(new BsonBinaryWriter(bsonOutput, payloadFieldNameValidator), bsonOutput, getSettings(),
-                        commandStartPosition, payload);
+                        startPosition, payload);
```

Comment:
Should you change the parameter name of the writePayload method from commandStartPosition to messageStartPosition to reflect that a different value is being passed in?

Reply:
Done.  And this also caught an inconsistency in BsonWriterHelper#writePayloadArray and its caller, SplittablePayloadBsonWriter.  Addressed those too.